File: python/calculator.py
#Calculator app

#Function to add numbers
from random import choices


def add(x, y):
    return x + y

#Function to subtract numbers
def subtract(x, y):
    return x - y

#Function to multiply numbers
def multiply(x, y):
    return x * y

#Function to divide numbers
def divide(x, y):
    return x / y

#Function for modulus
def modulus(x, y):
    return x % y



print("Welcome to calculator.py, \n\nSelect an operation:")
print("\n1. Addition\n2.Subtraction\n3.Multiplication\n4.Division\n5.Modulus")

while True:
    #Get user input
    choice = input("Enter your choice: ")
    choices = ["1", "2", "3", "4", "5"]

    #Check if user input is valid
    if choice in choices:
        num1 = float(input("Enter first number: "))
        num2 = float(input("Enter second number: "))

        if choice == '1':
            print(f"{num1} + {num2} = {add(num1, num2)}")

        elif choice == '2':
            print(f"{num1} - {num2} = {subtract(num1, num2)}")

        elif choice == '3':
            print(f"{num1} * {num2} = {multiply(num1, num2)}")
    
        elif choice == '4':
            print(f"{num1} / {num2} = {divide(num1, num2)}")
            
        elif choice == '5':
            print(f"{num1} % {num2} = {modulus(num1, num2)}")
            
        break
    else:
        print("Invalid input")

File: frontend/Javascript/Calculator/calculator.js
// Calculator App

// Function to subtract numbers
function subtract(a, b) {
  return a - b;
}

// Function to add numbers
function add(a, b) {
  return a + b;
}

// Function to multiply numbers
function multiply(a, b) {
  return a * b;
}

// Function to divide numbers
function divide(a, b) {
  return a / b;
}

alert(
  "Welcome to the calculator app!\n\nThis app will help you calculate the result of your math problem."
);

alert(
  "Which operation would you like to perform?\n\n1. Add\n2. Subtract\n3. Multiply\n4. Divide"
);

while (true) {
  // Get user input
  let choice = prompt("\nEnter your choice : ");
  let choices = ["1", "2", "3", "4"];

  // Check if user input is valid
  if (choices.includes(choice)) {
    let a = parseInt(prompt("Enter first number: "));
    let b = parseInt(prompt("Enter second number: "));

    // Perform operation
    switch (choice) {
      case "1":
        alert(`${a} + ${b} = ${add(a, b)}`);
        break;
      case "2":
        alert(`${a} - ${b} = ${subtract(a, b)}`);
        break;
      case "3":
        alert(`${a} x ${b} = ${multiply(a, b)}`);
        break;
      case "4":
        alert(`${a} / ${b} = ${divide(a, b)}`);
        break;
    }
  } else {
    alert("Invalid choice. Please try again.");
  }
  break;
}
